costmap_2d/include/costmap_2d/costmap_2d.h

```diff
@@ -450,11 +450,11 @@ class Costmap2D
       for (unsigned int i = 0; i < end; ++i)
       {
         at(offset);
-        offset += offset_a;
+        offset += static_cast<unsigned int>(offset_a);
```

Comment:
tbh i would change the function signature instead. 
the signature of the function allows for a larger range than what the implementation expects which is not ideal for setting expectations for how to use the method